Remotion/Web/Core/UI/SmartPage.cs

```diff
@@ -543,6 +624,40 @@ public bool? EnableSmartFocusing
     set { _enableSmartFocusing = value; }
   }
 
+  /// <summary>
+  /// Gets or sets the flag that determines whether a Content-Security-Policy header is set.
+  /// The value of the header is determined by <see cref="GetCspHeader"/>.
+  /// </summary>
+  [Description("The flag that determines whether a Content-Security-Policy header is set.")]
+  [Category("Behavior")]
+  [DefaultValue(false)]
+  public bool IsCspEnabled
+  {
+    get
+    {
+      _isCspEnabled ??= SafeServiceLocator.Current.GetInstance<ICspDefaultsProvider>().GetDefaultIsCspEnabledForPage(this);
+      return _isCspEnabled.Value;
+    }
+    set { _isCspEnabled = value; }
+  }
```

Comment:
setable properties in webforms: they're special and we cannot asign them while getting. Actually, assigning them durign the get operation is problematic even in non-webforms code :)

the pattern used can be seen with EnableSmartFocusing: we got the settable property EnableSmartFocusing and the protected virtual readonly property IsSmartFocusingEnabled. The same pattern should be used here. (disclaimer: with SmartFocusing, it's also part of an interface, so we used protected instead of public. I think protected is still sensible, though...)

Reply:
I don't get the problem of settable properties. Could you elaborate on why this is a problem?

Is the `EnableSmartFocusing` just an example or do you want to introduce a virtual get method as well? I would keep just the property and assign the default values in the constructor using the defaults provider and call it a day.

Reply:
@a-ctor  the first reason is that getting the settable property changes the state of the object, which is something we should always avoid because it breaks the idomatic property concept. It's also really impure ;)

The second reason is related to how WebForms Designer works (well...worked). Properties that are persisted in the ASPX get persisted based on the final state they have. This means when the Designer reads the property, it will see a value different from the default and persist it. 

A third reason has to do with consistency. When we start using two wildy separate patterns in the same file, it becomes much more difficult to maintain. 

I believe that the first behavior alone is sufficient to justify the more elaborate style.